homeassistant/components/nexia/climate.py

```diff
@@ -136,6 +139,11 @@ async def async_setup_entry(
         SET_HVAC_RUN_MODE_SCHEMA,
         f"async_{SERVICE_SET_HVAC_RUN_MODE}",
     )
+    platform.async_register_entity_service(
+        SERVICE_SET_FAN_SPEED,
```

Comment:
Why don't add a fan entity instead?

Reply:
A `Number` entity seems like the middle way here then and not a custom service.

Reply:
That makes sense to me. I can make that change!

Reply:
Changes implemented!